content/engineering/processes/development-processes.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Development Processes"
+date: 2021-08-31T18:10:57+02:00
+draft: false
+description: "Development processes and coding standards"
+weight: 1
+---
+
+## Git
+
+- We follow a gitflow like strategy. All changes are made in branches based off of `main`
+- Branches are merged into `main` via PR
+- Frontend applications are deployed to individual sandbox environments for each pull request.
+- Frontend applications are automatically deployed to staging when merging to `main`
+- Backend applications are deployed to a dedicated staging environment when being merged to `main`
+- Backend applications are deployed to production when invoking the `release` workflow
+- We use _strict_ [semantic versioning](https://semver.org/) on _all_ packages. Please learn this and implement it
+  strictly
+- Below version `1.0.0`, the API is considered unstable
+
+## Commit Messages
+
+- use the imperative, present tense: "change" not "changed" nor "changes"
+- No capitals and no fullstops (.) at the end
+- When a PR is ready for merge we trigger a squash and merge which will squash all your commits into one, when doing that follow the pattern of `type/scope/ticket-number: description (#PR-number)`. i.e. `style/onboarding/PROJECT-123: change button color to red (#321)`.
+
+## Pull Requests
+
+- Unit & e2e tests and snapshot updates should be performed prior to pushing your branch to the repository. This limits the the number of unnecessary CI runs and is a time and cost saving measure. More commits, fewer pushes.
+- PR title must follow the pattern of `type/scope/ticket-number: description`. i.e. `style/onboarding/PROJECT-123: change button color to red`.
```

Comment:
@CodeNameNinja  this is different from current PR template that we use.

please check it out

Reply:
we should discsuss this tomorrow, I dont like the above pattern, but Paul would like us to stick to 1 format company wide